src/main/java/application/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,190 @@
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        User user = null;
+        String query = "SELECT * FROM users WHERE deleted = FALSE AND login = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get user with login " + login, e);
+        }
+        user.setRoles(getRoles(user.getId()));
+        return Optional.of(user);
+    }
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO users(name, login, password) VALUES (?, ?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong(1));
+            }
+            return insertRoles(user);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create user " + user, e);
+        }
+    }
+
+    @Override
+    public Optional<User> get(Long id) {
+        User user = null;
+        String query = "SELECT * FROM users INNER JOIN users_roles ON id = user_id "
+                + "WHERE deleted = FALSE AND id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get user with id " + id, e);
+        }
+        user.setRoles(getRoles(user.getId()));
+        return Optional.of(user);
+    }
+
+    @Override
+    public List<User> getAll() {
+        List<User> users = new ArrayList<>();
+        String query = "SELECT * FROM users INNER JOIN users_roles ur ON id = user_id "
```

Comment:
why do you need inner join if you fetching roles with separate query?

Reply:
oops. done